aws-cpp-sdk-core-tests/aws/auth/AWSHttpResourceClientTest.cpp

```diff
@@ -134,6 +164,63 @@ namespace
         ASSERT_EQ("/path/to/res", mockRequest.GetUri().GetPath());
         ASSERT_EQ(Aws::Http::HttpMethod::HTTP_GET, mockRequest.GetMethod());
         ASSERT_EQ("{ \"Resource\": \"TestResource\" }", result);
+        // No retries should be made
+        ASSERT_EQ(1u, mockHttpClient->GetAllRequestsMade().size());
+    }
+
+    TEST_F(AWSHttpResourceClientTest, TestAWSHttpResourceClientWithTemporaryNullResponse)
```

Comment:
What about `WithSuccessResponseAfterOneNullResponse`? This can help explain why it's `2` in the last ASSERT